sonic_data_client/mixed_db_client.go

```diff
@@ -571,98 +582,28 @@ func (c *MixedDbClient) handleTableData(tblPaths []tablePath) error {
 			}
 		}
 
-		if tblPath.operation == opRemove {
-			//Only table name provided
-			if tblPath.tableKey == "" {
-				// tables in COUNTERS_DB other than COUNTERS table doesn't have keys
-				if tblPath.dbName == "COUNTERS_DB" && tblPath.tableName != "COUNTERS" {
-					pattern = tblPath.tableName
-				} else {
-					pattern = tblPath.tableName + tblPath.delimitor + "*"
-				}
-				// Can't remove entry in temporary state table
-				dbkeys, err = redisDb.Keys(pattern).Result()
-				if err != nil {
-					log.V(2).Infof("redis Keys failed for %v, pattern %s", tblPath, pattern)
-					return fmt.Errorf("redis Keys failed for %v, pattern %s %v", tblPath, pattern, err)
-				}
-			} else {
-				// both table name and key provided
-				dbkeys = []string{tblPath.tableName + tblPath.delimitor + tblPath.tableKey}
-			}
+		if tblPath.tableKey == "" {
+			// Must provide table key
+			return fmt.Errorf("Unsupported path %v, can't update table", tblPath)
+		}
 
-			for _, dbkey := range dbkeys {
-				tableKey := strings.TrimPrefix(dbkey, tblPath.tableName + tblPath.delimitor)
-				err = c.DbDelTable(tblPath.tableName, tableKey)
-				if err != nil {
-					log.V(2).Infof("swsscommon delete failed for  %v, dbkey %s", tblPath, dbkey)
-					return err
-				}
+		if tblPath.operation == opRemove {
+			err = c.DbDelTable(tblPath.tableName, tblPath.tableKey)
```

Comment:
The delete multiple keys code seems been removed?

Reply:
yes

Reply:
Thanks, client still can delete multiple keys with 1 gnmi API call: add multiple delete operation in 1 APi call, every delete operation will delete 1 key.